src/dynamic_scene_graph.cpp

```diff
@@ -469,105 +367,78 @@ bool DynamicSceneGraph::removeNode(NodeId node_id) {
 }
 
 bool DynamicSceneGraph::removeEdge(NodeId source, NodeId target) {
-  LayerKey source_key, target_key;
-  if (!hasEdge(source, target, &source_key, &target_key)) {
-    return false;
-  }
-
-  if (!source_key.valid() || !target_key.valid()) {
+  const auto lookup = lookupEdge(source, target);
+  if (!lookup.exists) {
+    // also excludes invalid edges
     return false;
   }
 
-  if (source_key == target_key) {
-    return layerFromKey(source_key).removeEdge(source, target);
+  if (lookup.isSameLayer()) {
+    return layerFromKey(lookup.source).removeEdge(source, target);
   }
 
-  removeInterlayerEdge(source, target, source_key, target_key);
+  removeInterlayerEdge(source, target, lookup.source, lookup.target);
   return true;
 }
 
-bool DynamicSceneGraph::isDynamic(NodeId source) const {
-  auto iter = node_lookup_.find(source);
-  if (iter == node_lookup_.end()) {
-    return false;
-  }
-
-  return iter->second.dynamic;
-}
-
 size_t DynamicSceneGraph::numLayers() const {
   const size_t static_size = layers_.size();
 
-  size_t unique_dynamic_layers = 0;
-  for (const auto& id_layer_group_pair : dynamic_layers_) {
-    if (!layers_.count(id_layer_group_pair.first)) {
-      unique_dynamic_layers++;
+  size_t unique_layer_groups = 0;
+  for (const auto& [layer_id, partitions] : layer_partitions_) {
+    if (!layers_.count(layer_id)) {
+      ++unique_layer_groups;
     }
   }
 
-  return static_size + unique_dynamic_layers;
-}
-
-size_t DynamicSceneGraph::numDynamicLayersOfType(LayerId layer) const {
-  if (!dynamic_layers_.count(layer)) {
-    return 0;
-  }
-
-  return dynamic_layers_.at(layer).size();
-}
-
-size_t DynamicSceneGraph::numDynamicLayers() const {
-  size_t num_layers = 0;
-  for (const auto& id_layer_group : dynamic_layers_) {
-    num_layers += id_layer_group.second.size();
-  }
-  return num_layers;
-}
-
-size_t DynamicSceneGraph::numNodes(bool include_mesh) const {
-  return numStaticNodes() + numDynamicNodes() +
-         ((mesh_ == nullptr || !include_mesh) ? 0 : mesh_->numVertices());
+  return static_size + unique_layer_groups;
 }
 
-size_t DynamicSceneGraph::numStaticNodes() const {
-  size_t total_nodes = 0u;
-  for (const auto& id_layer_pair : layers_) {
-    total_nodes += id_layer_pair.second->numNodes();
+size_t DynamicSceneGraph::numNodes() const {
+  size_t total_nodes = numUnpartitionedNodes();
+  for (const auto& [layer_id, partitions] : layer_partitions_) {
+    for (const auto& [partition_id, partition] : partitions) {
+      total_nodes += partition->numNodes();
+    }
   }
 
   return total_nodes;
 }
 
-size_t DynamicSceneGraph::numDynamicNodes() const {
+size_t DynamicSceneGraph::numUnpartitionedNodes() const {
```

Comment:
What does unpartitioned nodes mean? This looks like just num nodes in DSG to me?